x-pack/plugins/cloud_security_posture/public/common/api/create_detection_rule.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { HttpSetup } from '@kbn/core/public';
+
+const DETECTION_ENGINE_URL = '/api/detection_engine' as const;
+const DETECTION_ENGINE_RULES_URL = `${DETECTION_ENGINE_URL}/rules` as const;
+
+interface RuleCreateProps {
```

Comment:
Thanks @JordanSh for pointing this out, unfortunately, that's something we can't do, at least for now, I initially tried reusing all the variables and types, as that adds some safety for us to know when the signature or contract changes, but I got this error on [the CI](https://buildkite.com/elastic/kibana-pull-request/builds/145739#01899f34-7bfd-4d8c-95f9-4877d96192a3):

```
 proc [tsc] error TS6202: Project references may not form a circular graph. Cycle detected: /var/lib/buildkite-agent/builds/kb-n2-4-spot-462fd903344e64ea/elastic/kibana-pull-request/kibana/tsconfig.refs.json
```

This error happens because the `security_solution` plugin imports some files from the `cloud_security_plugin`, so once  `cloud_security_plugin` imports anything from the `security_solution` plugin (even if it's only types files) this Circular dependency error happens, leading to an infinite recursion on the compiler.

I researched only a couple of options to workaround circular dependency:

1 - Move and refactor the whole rules into their own plugin, so `security_solution` and `cloud_security_plugin` plugins can import without causing a dependency. However, that's not feasible due to the large refactor required.
2 - Dependency injection: make the `security_solution` plugin consume the `cloud_security_plugin` while passing as an initialization parameter all the dependencies needed (the `createRule` function for example), however, this approach doesn't work for types, so for us to have type checking as well, the solution would be to move all the types to a shared package (ex: @detection-security-rules), where the `security_solution` and `cloud_security_plugin` plugins can import without causing a dependency. This is feasible, however, it takes more time to move the types to a common package. 
3 - Duplication: simply duplicate the logic and types, since it's mainly an API call and it's API return Type, I went with that option, since that sounded to me pretty plausible to be done in the short time for an MVP version.

If that makes sense we can set Option 2 (dependency injection + separate package for types) as a goal and add it to the technical debts list, I think we can probably request the types to be in a separate package from the maintainer team.